.github/workflows/build.yml

```diff
@@ -78,93 +110,193 @@ jobs:
-      #        run: |-
-      #          ./mvnw -B -Dstyle.color=always \
-      #          -Dtycho.mode=maven  \
-      #          -P update-target-versions \
-      #          -f ./targetdefinition/pom.xml \
-      #          validate
-      #
-      #      - name: Update category
-      #        run: |-
-      #         ./mvnw -B -Dstyle.color=always \
-      #          -Dtycho.mode=maven \
-      #          -P update-category-versions \
-      #          -f ./site/pom.xml \
-      #          validate
-      #
-      #      - name: Set tycho version
-      #        run: |-
-      #          ./mvnw -B -Dstyle.color=always \
-      #          -Dtycho.mode=maven  \
-      #          -P update-category-versions \
-      #          tycho-versions:update-eclipse-metadata
-      #
-      - name: Build with Maven
+        # Eclipse specific build block. BEGIN
+        # It manages tycho versions in P2 world.
+        # Different eclipse project target / category can be in different places
+        #      - name: Set project version
+        #        run: |-
+        #          ./mvnw -B -Dstyle.color=always \
+        #          -DnewVersion=${{ steps.version.outputs.version }} \
+        #          -DgenerateBackupPoms=false \
+        #          -Dtycho.mode=maven \
+        #          versions:set
+        #
+        #      - name: Update target definition
+        #        run: |-
+        #          ./mvnw -B -Dstyle.color=always \
+        #          -Dtycho.mode=maven  \
+        #          -P update-target-versions \
+        #          -f ./targetdefinition/pom.xml \
+        #          validate
+        #
+        #      - name: Update category
+        #        run: |-
+        #         ./mvnw -B -Dstyle.color=always \
+        #          -Dtycho.mode=maven \
+        #          -P update-category-versions \
+        #          -f ./site/pom.xml \
+        #          validate
+        #
+        #      - name: Set tycho version
+        #        run: |-
+        #          ./mvnw -B -Dstyle.color=always \
+        #          -Dtycho.mode=maven  \
+        #          -P update-category-versions \
+        #          tycho-versions:update-eclipse-metadata
+        # Eclipse specific build block. END
+
+        # Phase 1 - Build and deploy to local stage (/target/stage
+      - name: Build with Maven (build, test)
+        run: |-
+          ./mvnw -B -Dstyle.color=always \
+          -Drevision=${{ steps.version.outputs.version }} \
+          -Psign-artifacts \
+          -Prelease-local-stage \
+          install
+
+      # Phase 2 - Deploy to judo-ng nexus
+      - name: Deploy with Maven (judo-ng)
         run: |-
           ./mvnw -B -Dstyle.color=always \
           -Drevision=${{ steps.version.outputs.version }} \
-          -Psign-artifacts,release-judong \
+          -DdeployOnly \
+          -Prelease-judong \
           deploy
-      #      - name: Release P2
+
+      # For maven central repository release
```

Comment:
It is release to CENTRAL repo, so this artifact have to be release. But some condition missing in the commented out block, which is a guard that the central release have to be run only when release version is presented. Have to check it is on a release/* branch

          ${{ always() && 
          contains(github.head_ref, 'release') &&
         job.status == 'success' }}